docs/guides/transfer_learning_full.md

```diff
@@ -0,0 +1,370 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.13.8
+kernelspec:
+  display_name: 'Python 3.8.11 (''.venv'': venv)'
+  language: python
+  name: python3
+---
+
+# Transfer Learning
+
+
+### Ideas
+* Dataset: use a simple dataset from TensorFlow Dataset like MNIST, KMNIST, or FashionMNIST, or even cats vs dogs which is a little bigger.
+* Model Options: 
+  * `jax-resnet`: has the benefit of being small and fast.
+  * `FlaxCLIPVisionModel`: would showcase how to use HuggingFace models.
+
+### Important Links
+- https://github.com/google/flax/discussions/1706
+- https://github.com/google/flax/discussions/2245
+
+### Outline
+* Introduction
+* Creating the model
+  * Loading pretrained module
+  * Creating a model for the new task
+  * Transfering pretrained parameters
+* Optimization Procedure
+  * Discuss Options
+    * Using `stop_gradient`
+    * Taking gradients w.r.t. trainable parameters only
+    * Zeroing out gradients for non-trainable parameters
+  * Using Optax `multi_transform` + `variables_path_partitions`
+    * Completely freezing layers
+    * Using different learning rates
+* Training the model
+
++++
+
+## Introduction
+
+### What is Transfer Learning?
+Theory
```

Comment:
I wouldn't use bullets here, also the separation between "theory" and "practice" is a bit arbitrary, why is "Pretrain on a base task, use the learned parameters to train on a target task" theory and "Fine-tuning: use the pretrained weights as a starting point but train the whole model on the new task." practice?

Please also add some sources, could be articles but also (good) blog posts where concepts of transfer learning are discussed.